src/Formats/FormatSettings.h

```diff
@@ -207,12 +207,20 @@ struct FormatSettings
         std::string format_schema;
         std::string format_schema_path;
         bool is_server = false;
-        /// proton: starts
-        std::string kafka_schema_registry_url;
-        std::string kafka_schema_registry_credentials;
-        /// proton: ends
     } schema;
 
+    /// proton: starts
+    struct
+    {
+        std::string url;
+        std::string credentials;
+        std::string private_key_file;
+        std::string certificate_file;
+        std::string ca_location;
+        bool skip_cert_check = false;
+    } kafka_schema_registry;
```

Comment:
We probably need a new name for the variable

Reply:
I think we should stick with `kafka_schema_registry` because it's not a generic schema registry, there are other schema registries too.

For the `_config` or `_context` suffix, I think it's redundant, it's already in `FormatSettings`, `format_settings.kafka_schema_registry` means kafka schema registry related settings already, I don't think we need a suffix. Also, there are other structs in `FormatSettings`, none of them has such suffix. Unless you have a strong concern, I think we should just keep it as it is.